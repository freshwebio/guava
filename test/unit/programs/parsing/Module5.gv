module Order exposing { create }

function create() :: Integer {
  3043923
}