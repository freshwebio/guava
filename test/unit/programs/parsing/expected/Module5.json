{
  "description": "should parse a program with a function that returns an integer constant with a single value inline type definition",
  "module": {
    "name": "Order",
    "exposes": {
      "exposed": ["create"]
    },
    "moduleStatements": {
      "statementList": [{
        "kind": "function",
        "fncStatement": {
          "kind": "functionWithInlineTypeDef",
          "functionName": "create",
          "defArgs": {
            "args": []
          },
          "functionTypeDefValueTypes": [{
            "kind": "valueTypeSingle",
            "valueTypeName": "Integer"
          }],
          "functionExpr": {
            "kind": "arithmeticExpr",
            "arithmeticExpr": {
              "kind": "integerConst",
              "intConstValue": 3043923
            }
          }
        }
      }]
    }
  }
}